enumerable_method.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # my_each
+  def my_each
+    return to_enum unless block_given?
+
+    each { |i| yield(i) }
+  end
+  # Self represent the current object, returns the elements of the array
+  # Yield asks for a block
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  # my_select
+  def my_select
+    # select: Returns a new array containing all elements of enum for which the given block returns a true value.
+    return to_enum unless block_given?
+
+    # new array
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
+
+  # my_all?
+
+  def my_all?
+    # all: The method returns true if ALL elements are true (or empty array).
+
+    my_each { |x| return false unless yield(x) }
+    true
+  end
+
+  # my_any?
+  def my_any?
+    # any: The method returns true if AT LEAST one element is true (or non empty array).
+
+    my_each { |x| return true if yield(x) }
+    false
+  end
+
+  # my_none?
```

Comment:
**[REQUIRED]**
1. when no block or argument is given returns true only if none of the collection members is true

2. when a class is passed as an argument returns true if none of the collection is a member of such class

3. when a Regex is passed as an argument returns true only if none of the collection matches the Regex

4. when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern